### PR TITLE
Fix #151: Add step lookup by step ID

### LINKED ISSUE #151: Add step lookup by step ID
Currently, we allow lookup by the step name. We should also add a method for lookup by a step ID.

### PR DESCRIPTION
**Pull request by petrdvorak**

### CODE REVIEW

File: powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/logging/ObjectStepLogger.java [hunk 1]
```diff
@@ -255,12 +255,26 @@ public void writeDoneFailed(String id) {
...
     }
 
+    /**
+     * Get first {@link StepItem} with given ID.
+     * @param id Item ID.
+     * @return First {@link StepItem} with given id or {@code null} if logger doesn't contain such item.
+     */
+    public StepItem getFirstItem(String id) {
+        for (StepItem item: items) {
+            if (item.getId().equals(id)) {
```
Comment (romanstrobl) [inline powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/logging/ObjectStepLogger.java:265]: Are we should a null pointer exception is not possible in this condition?

File: powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/logging/ObjectStepLogger.java [hunk 2]
```diff
@@ -277,12 +291,26 @@ public StepItem getFirstItem(String itemName) {
...
     }
 
+    /**
+     * Get first {@link StepError} with given name.
+     * @param id Error ID.
+     * @return First {@link StepError} with given ID or {@code null} if logger doesn't contain such error.
+     */
+    public StepError getFirstError(String id) {
+        for (StepError error: errors) {
+            if (error.getId().equals(id)) {
```
Comment (romanstrobl) [inline powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/logging/ObjectStepLogger.java:301]: dtto

File: powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/logging/ObjectStepLogger.java (hunk 1 above)
Comment (petrdvorak) [reply powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/logging/ObjectStepLogger.java:265]: I took this code from the previously available methods that were checking by name and gave this some thought and yes, we are sure there cannot be NPE.

The `StepItem` and `StepError` do not have setters, only the constructor with all required parameters. We are the only ones creating these instances and we always assign an ID and name. We can add a null check, it won't solve much...

Comment (petrdvorak) [reply powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/logging/ObjectStepLogger.java]: I inverted the conditions and added null checks.

File: powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/logging/ObjectStepLogger.java (hunk 2 above)
Comment (petrdvorak) [reply powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/logging/ObjectStepLogger.java]: I inverted the conditions and added null checks.

**Pull request merged**